lib/select/select.tsx

```diff
@@ -0,0 +1,336 @@
+        event.preventDefault();
+        event.stopPropagation();
+        const previousSibling = target.previousSibling as HTMLElement;
+        previousSibling.focus();
+      }
+
+      // 'ArrowDown' key
+      if (event.keyCode === 40) {
+        event.preventDefault();
+        event.stopPropagation();
+        const nextSibling = target.nextSibling as HTMLElement;
+        nextSibling.focus();
+      }
+    }
+  };
+
+  handleMouseDown = (event: React.MouseEvent | React.TouchEvent): void => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const selectRef: HTMLElement = this.selectRef.current;
+    const optionsRef: HTMLUListElement = this.optionsRef.current;
+
+    const target: HTMLElement = event.target as HTMLElement;
+
+    // select
+    if (target.className === selectRef.className) {
+      if (this.state.active) {
+        this.setState({ active: false });
+        target.focus();
+      } else {
+        this.setState({ active: true });
+        this.focusOptionByValue(optionsRef, this.state.value);
+      }
+
+      // options
+    } else if (target.className === namespace('select__item')) {
+      this.setState(
+        {
+          active: false,
+          value: target.getAttribute('data-item')
+        },
+        () => {
+          this.handleChange();
+        }
+      );
+
+      selectRef.focus();
+    }
+  };
+
+  handleBlur = (event: React.FocusEvent): void => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const target: HTMLElement = event.target as HTMLElement;
+    const relatedTarget: HTMLElement =
+      (event.relatedTarget as HTMLElement) ||
+      (document.activeElement as HTMLElement);
+
+    if (
+      (relatedTarget !== null &&
+        target.getAttribute('data-id') !==
+          relatedTarget.getAttribute('data-id')) ||
+      relatedTarget === null
+    ) {
+      this.setState({
+        active: false
+      });
+    }
+  };
+
+  getLabelFromValue = (value: string): string => {
+    const { options } = this.props;
+
+    let label = '';
+
+    for (const option of options) {
+      if (option.value === value) {
+        label = option.label;
+        break;
+      }
+    }
+
+    return label;
+  };
+
+  focusOptionByValue = (options: HTMLUListElement, value: string): void => {
+    if (value) {
+      for (const child of Array.from(options.childNodes)) {
+        const element = child as HTMLElement;
+        if (element.getAttribute('data-item') === this.state.value) {
+          element.focus();
+          break;
+        }
+      }
+    } else {
+      const option = options.firstChild as HTMLElement;
+      option.focus();
+    }
+  };
+
+  render() {
+    const {
+      props,
+      state,
+      id,
+      selectRef,
+      optionsRef,
+      handleChange,
+      handleKeyDown,
+      handleMouseDown,
+      handleBlur,
+      getLabelFromValue
+    } = this;
+
+    const bem = BEM('select');
+    bem.addModifiers(state.error ? 'error' : '');
```

Comment:
I actually thought it would be better to short-circuit these, rather than ternary:`bem.addModifiers(state.error ? 'error' : '')` --> `state.error && bem.addModifiers('error')`

Reply:
We need to review our TypeScript config for this one. I get warning when trying to short-circuit.

Reply:
https://github.com/airbnb/javascript#control-statements--value-selection
I usually find short-circuiting is obscuring the code